### ISSUE TITLE
[ramda_v0.26.x] the head can return undefined value

### ISSUE THREAD

**Issue by FilipMessa** [flow-typed/flow-typed#3355]
The `head` function can returns the undefined.

```R.head([]); //=> undefined``` (https://ramdajs.com/docs/#head)

But actual the types are written like this:

line 649: `declare function head<T, V: Array<T>>(xs: V): $ElementType<V, 0>;`

I expect that it should be something like this.
```
declare function head<T, V: Array<T>>(xs: V): ?$ElementType<V, 0>;
```

**Comment by LoganBarnett** [comment]
I think `$ElementType` covers this. If you look at the test:

``` javascript
 describe('head', () => {
    describe('with array', () => {
      it('should returns MaybeOf element type', () => {
        const fn = (arr: number[]) => {
          const xOfXs: ?number = _.head(arr);
          //$ExpectError
          const xOfXs: number = _.head(arr);
        }
      });
...
```

Also you can do a [try](https://flow.org/try/#0CYUwxgNghgTiAEAzArgOzAFwJYHtXwAsQpgAeAFQBp4A1ALngEEYYoBPCgPk4AoAPAM4MaASgYASAKIQQAWxCoM5NgAcQpGtQAMnANwAofWDwCMSHDgYA3HFmDwAvIWLAeAbQC6IoA) to show it with an empty array directly:

``` javascript
declare function head<T, V: Array<T>>(xs: V): $ElementType<V, 0>;

const foo: void = head([])
```

TLDR; I feel like we can close this unless we can get some sort of reproduction that `head` isn't doing the expected thing.

**Comment by AndrewSouthpaw** [comment]
Agreed, thanks @LoganBarnett.
